lib/CGameInfoCallback.cpp

```diff
@@ -273,6 +273,16 @@ bool CGameInfoCallback::getHeroInfo(const CGObjectInstance * hero, InfoAboutHero
 
 	bool accessFlag = hasAccess(h->tempOwner);
 
+	if (!accessFlag && gs->curB) //if it's battle we can get enemy hero full data
```

Comment:
Fact that some battle is going is not enough to provide info like that.

What I think would even better is to move this check into CBattleInfoEssentials callback and name it like "CBattleInfoEssentials::playerHasAccessToHeroInfo".

Then in this case it's would be:
`if (!accessFlag && gs->curB)
    accessFlag = gs->curB->playerHasAccessToHeroInfo();`